pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,433 @@
+use drink::{
+	sandbox_api::assets_api::AssetsAPI,
+	session::{error::SessionError, Session},
+};
+use frame_support::assert_ok;
+use pop_api::{
+	primitives::TokenId,
+	v0::fungibles::events::{Approval, Created, Transfer},
+};
+use pop_sandbox::{Balance, DevnetSandbox as Sandbox, ALICE, BOB};
+use scale::Encode;
+use utils::*;
+
+use super::*;
+
+const TOKEN_ID: TokenId = 1;
+const TOKEN_MIN_BALANCE: Balance = 10_000;
+
+#[drink::contract_bundle_provider]
+enum BundleProvider {}
+
+#[drink::test(sandbox = Sandbox)]
+fn new_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	// Token exists after the deployment.
+	assert!(session.sandbox().asset_exists(&TOKEN_ID));
+	let expected = Created {
+		id: TOKEN_ID,
+		creator: account_id_from_slice(contract.as_ref()),
+		admin: account_id_from_slice(contract.as_ref()),
+	}
+	.encode();
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice()); // Successfully emit event.
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn existing_constructor_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+
+	// Fails to deploy contract with a non-existing token ID.
+	assert!(
+		deploy_with_existing_constructor(&mut session, BundleProvider::local()?, TOKEN_ID).is_err()
+	);
+	expect_deployment_reverted(&session, PSP22Error::Custom(String::from("Unknown")));
+
+	// Successfully deploy contract with an existing token ID.
+	let actor = session.get_actor();
+	session.sandbox().create(&TOKEN_ID, &actor, TOKEN_MIN_BALANCE).unwrap();
+	deploy_with_existing_constructor(&mut session, BundleProvider::local()?, TOKEN_ID)?;
+
+	assert!(session.sandbox().asset_exists(&TOKEN_ID));
+	assert_eq!(last_contract_event(&session), None); // No event emitted.
+	assert_eq!(session.sandbox().total_supply(&TOKEN_ID), 0);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn balance_of_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+
+	const AMOUNT: Balance = TOKEN_MIN_BALANCE * 4;
```

Comment:
Lets move this up with the other constants so that you don't have to define it multiple times.